base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.cpp

```diff
@@ -241,10 +259,20 @@ Result<Json::Value> StatusFetcher::FetchGroupStatus(
                                  original_request.FileDescriptors()};
   auto [_, instances_json, group_response] =
       CF_EXPECT(FetchStatus(group_request));
-  CF_EXPECT(
-      group_response.status().code() == cvd::Status::OK,
-      fmt::format("Running `cvd status --all_instances` for group \"{}\" failed",
-                  group.GroupName()));
+  if (group_response.status().code() != cvd::Status::OK) {
+    group.SetAllStates(cvd::InstanceState::UNREACHABLE);
+    CF_EXPECT(instance_manager_.UpdateInstanceGroup(group));
+    instances_json = Json::Value(Json::arrayValue);
+    for (const auto& instance: group.Instances()) {
+      Json::Value instance_json;
+      instance_json["instance_name"] = instance.PerInstanceName();
+      instances_json.append(instance_json);
+    }
+  }
+  for (size_t i = 0; i < group.Instances().size(); ++i) {
+    instances_json[(int)i]["status"] =
+        cvd::InstanceState_Name(group.Instances()[i].State());
+  }
```

Comment:
Would it be possible to break up this method into two methods based on the two large behaviors it has?

1. Update `selector::LocalInstanceGroup` state based on `cvd status`
2. Produce a JSON report representing the `selector::LocalInstanceGroupState`

If there's a lot of `cvd status` output not captured in `selector::LocalInstanceGroupState`, is it possible to add more fields to capture that additional state?

Reply:
Doesn't apply after addressing the previous comment. Move this to `FetchOneInstanceStatus` which almost everything else calls here. Updating the instance state there is two lines, so probably doesn't make much sense to extract.

Reply:
It still feels somewhat unrelated to the rest of the behavior of this function and I'm generally not a fan of output parameters, but it also looks inconvenient to extract.

Also, `FetchStatus` calls `FetchOneInstanceStatus` which calls `UpdateInstance` in the same unreachable state, is it necessary to do the same here?

Reply:
Fixed